.github/actions/deploy-happy-stack/action.yml

```diff
@@ -49,11 +48,16 @@ inputs:
     description: "Optional stack to pull images from rather than building them (i.e. should be used with image_source_env)"
     required: false
     default: ""
+  github-repo-branch:
+    description: "Branch of Github repo to deploy"
+    required: false
 
 runs:
   using: "composite"
   steps:
     - uses: actions/checkout@v3
+      with:
+          ref: ${{ inputs.github-repo-branch || github.event.pull_request && github.head_ref || github.ref_name }}
```

Comment:
For context, the reason this isn't more straightfoward is here: https://github.com/github/docs/issues/15319#issuecomment-1620624403